common/src/primitives/signed_amount.rs

```diff
@@ -0,0 +1,1039 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(clippy::eq_op)]
+
+use std::iter::Sum;
+
+use super::Amount;
+
+pub type SignedIntType = i128;
+
+/// A signed fixed-point type for amounts used in accounting, specifically
+/// The smallest unit of count is called an atom
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SignedAmount {
+    val: SignedIntType,
+}
+
+fn remove_right_most_zeros_and_decimal_point(s: String) -> String {
+    let point_pos = s.chars().position(|c| c == '.');
+    if point_pos.is_none() {
+        return s;
+    }
+    let s = s.trim_end_matches('0');
+    let s = s.trim_end_matches('.');
+    s.to_owned()
+}
+
+impl SignedAmount {
+    pub const MAX: Self = Self::from_atoms(SignedIntType::MAX);
+
+    pub const fn from_atoms(v: SignedIntType) -> Self {
+        SignedAmount { val: v }
+    }
+
+    pub fn into_atoms(&self) -> SignedIntType {
+        self.val
+    }
+
+    pub fn from_unsigned(amount: Amount) -> Option<Self> {
+        let unsigned_atoms = amount.into_atoms();
+        let atoms: SignedIntType = unsigned_atoms.try_into().ok()?;
+        Some(Self::from_atoms(atoms))
+    }
+
+    pub fn into_unsigned(self) -> Option<Amount> {
+        let atoms = self.val;
+        let unsigned_atoms: super::amount::UnsignedIntType = atoms.try_into().ok()?;
+        Some(Amount::from_atoms(unsigned_atoms))
+    }
+
+    pub fn into_fixedpoint_str(self, decimals: u8) -> String {
```

Comment:
I like the idea of reusing `Amount::from_fixedpoint_str`. Can we do the same for `into_fixedpoint_str`? Not not only we reuse the logic but also can get rid of duplicated `remove_right_most_zeros_and_decimal_point` in this file

Reply:
Done